### DISCUSSION TITLE
auto-pairs: underscore `_`

### DISCUSSION THREAD

**Discussion by Indyandie** [helix-editor/helix#8898] [Troubleshooting] [answered]
### Summary

It seems that underscores (`_`) don't seem to work for pairing. 

### Reproduction Steps

 

I tried this:

1. updated my `config.toml`

```toml
[editor.auto-pairs]
'(' = ')'
'{' = '}'
'[' = ']'
'"' = '"'
'`' = '`'
'<' = '>'
'*' = '*'
"_" = "_"
'~' = '~'
```

I expected an auto pair (`__`) when I tap the `_` key

> The `*` and `~` auto pair as expected. 

Instead, a single `_` appears. 

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
no relevant logs
```

</details>

### Platform

macos

### Terminal Emulator

alacritty

### Installation Method

brew

### Helix Version

helix 23.10

**Comment by the-mikedavis** [comment, marked as answer]
I can't reproduce this unless I switch to a language that already has custom auto-pairs set, for example Rust: https://github.com/helix-editor/helix/blob/090ed97e0045bfad1e5bff8b96c61707b996b85a/languages.toml#L183-L199

For that case you should add new auto-pairs like `_` to that language config.

**Reply by Indyandie** [reply]
🙇 Thanks for confirming @the-mikedavis and my apologies looks like it's working as expected. Not sure if it was a reboot or if I'm just blind.

To your point on adding the `auto-pairs` to the language config, that was my first attempt using markdown. Unfortunately, that didn't seem to work and I took the `editor.auto-pairs` route.

```toml
[[language]]
name = "markdown"
scope = "source.markdown"

[language.auto-pairs]
"_" = "_"
'*' = '*'
```
